crates/server/src/domain/quests.rs

```diff
@@ -95,25 +95,22 @@ pub async fn get_instance_state(
     quest_id: &str,
     quest_instance: &str,
 ) -> Result<(Quest, QuestState), QuestError> {
-    let quest = db.get_quest(quest_id).await;
-    match quest {
-        Ok(stored_quest) => {
-            let quest = stored_quest.to_quest()?;
-            let stored_events = db.get_events(quest_instance).await?;
+    let quest = db.get_quest(quest_id).await.map_err(|_| {
+        QuestError::CommonError(CommonError::BadRequest(
+            "the quest instance ID given doesn't correspond to a valid quest".to_string(),
+        ))
+    })?;
```

Comment:
Why don't we do `db.get_quest(quest_id).await?`?  because the `From<DBError>` implementation should handle 500 or 400. Instead, here, we're always returning a BadRequest.